tests/bloom_filter/filter_bitset_test.cu

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (c) 2024, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <test_utils.hpp>
+
+#include <cuco/bloom_filter.cuh>
+
+#include <cuda/functional>
+#include <thrust/device_vector.h>
+#include <thrust/functional.h>
+
+#include <catch2/catch_template_test_macros.hpp>
+#include <catch2/generators/catch_generators.hpp>
+
+#include <random>
+#include <type_traits>
+
+namespace {
+
+template <typename Key>
+thrust::device_vector<uint32_t> get_arrow_filter_reference_bitset()
+{
+  static std::vector<thrust::device_vector<uint32_t>> const reference_bitsets{
```

Comment:
Generated by extracting bloom filter and xxhash implementations from Arrow and throwing them together with a main here: https://godbolt.org/z/ebdddezbP